runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XPPanel.java

```diff
@@ -0,0 +1,157 @@
+
+		setMinimumSize(new Dimension(PANEL_WIDTH, PANEL_HEIGHT));
+		setPreferredSize(new Dimension(PANEL_WIDTH, PANEL_HEIGHT));
+		setSize(PANEL_WIDTH, PANEL_HEIGHT);
+		setVisible(true);
+		statsPanel = new JPanel();
+		statsPanel.setLayout(new GridLayout(24, 1));
+
+		try
+		{
+			for (Skill skill : Skill.values())
+			{
+				if (skill == Skill.OVERALL)
+					break;
+
+				JLabel skillLabel = new JLabel();
+				labelMap.put(skill, makeSkillPanel(skill, skillLabel));
+			}
+		} catch (IOException e)
+		{
+			e.printStackTrace();
+
+		}
+		JButton resetButton = new JButton("Reset All");
+		resetButton.setPreferredSize(new Dimension(PANEL_WIDTH, 32));
+		resetButton.addActionListener((ActionEvent e) ->
+				runelite.getExecutor().execute(this::resetAllSkillXpHr));
+
+		statsPanel.add(resetButton);
+		JScrollPane scroll = new JScrollPane(statsPanel);
+		scroll.add(statsPanel);
+
+		add(statsPanel);
+	}
+
+	private JButton makeSkillResetButton(Skill skill) throws IOException
+	{
+		ImageIcon resetIcon = new ImageIcon(ImageIO.read(getClass().getResourceAsStream("reset.png")));
+		JButton resetButton = new JButton(resetIcon);
+		resetButton.setPreferredSize(new Dimension(32, 32));
+		resetButton.addActionListener(actionEvent -> resetSkillXpHr(skill));
+		return resetButton;
+	}
+
+	private JPanel makeSkillPanel(Skill skill, JLabel levelLabel) throws IOException
+	{
+		BorderLayout borderLayout = new BorderLayout();
+		borderLayout.setHgap(12);
+		JPanel iconLevel = new JPanel(borderLayout);
+		iconLevel.setPreferredSize(new Dimension(PANEL_WIDTH, 32));
+
+		String skillIcon = "/skill_icons/" + skill.getName().toLowerCase() + ".png";
+		logger.debug("Loading skill icon from {}", skillIcon);
+		JLabel icon = new JLabel(new ImageIcon(ImageIO.read(XPPanel.class.getResourceAsStream(skillIcon))));
+		iconLevel.add(icon, BorderLayout.LINE_START);
+
+		iconLevel.add(levelLabel, BorderLayout.CENTER);
+		iconLevel.add(makeSkillResetButton(skill), BorderLayout.LINE_END);
+
+		return iconLevel;
+	}
+
+	public void resetSkillXpHr(Skill skill)
+	{
+		xpTracker.getXpInfos()[skill.ordinal()].reset();
+		statsPanel.remove(labelMap.get(skill));
+		statsPanel.revalidate();
+		statsPanel.repaint();
+	}
+
+	public void resetAllSkillXpHr()
+	{
+		for (SkillXPInfo skillInfo : xpTracker.getXpInfos())
+		{
+			if (skillInfo != null && skillInfo.getSkillTimeStart() != null)
+				resetSkillXpHr(skillInfo.getSkill());
+		}
+	}
+
+	public void updateAllSkillXpHr()
+	{
+		for (SkillXPInfo skillInfo : xpTracker.getXpInfos())
+		{
+			if (skillInfo != null && skillInfo.getSkillTimeStart() != null)
+				updateSkillXpHr(skillInfo);
+		}
+	}
+
+	public void updateSkillXpHr(SkillXPInfo skillXPInfo)
+	{
+		JPanel skillPanel = labelMap.get(skillXPInfo.getSkill());
+		JLabel xpHr = (JLabel) skillPanel.getComponent(1);
+		xpHr.setText(NumberFormat.getInstance().format(skillXPInfo.getXpHr()) + " xp/hr");
+
+		if (skillPanel.getParent() != statsPanel)
```

Comment:
When a skill gets reset, I remove it from the stats panel, instead of having to remake the skill JPanel. This makes sure that if its already shown on the plugin, it will not add it again.